File: src/go/ratom/module_rate.go
package ratom

import (
	"context"
	"fmt"

	// "io"
	"time"

	"cloud.google.com/go/storage"
)

func SetMetadata(bucket, object string, r *Repo) error {
	ctx := context.Background()
	client, err := storage.NewClient(ctx)
	if err != nil {
		return fmt.Errorf("storage.NewClient: %v", err)
	}
	defer client.Close()

	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
	defer cancel()

	o := client.Bucket(bucket).Object(object)

	attrs, err := o.Attrs(ctx)
	if err != nil {
		return fmt.Errorf("object.Attrs: %v", err)
	}
	o = o.If(storage.Conditions{MetagenerationMatch: attrs.Metageneration})

	objectAttrsToUpdate := storage.ObjectAttrsToUpdate{
		Metadata: map[string]string{
			"NET_SCORE": fmt.Sprintf("%f", r.NetScore), 
			"RESPONSIVE_MAINTAINER": fmt.Sprintf("%f", r.Responsiveness),
			"RAMP_UP" : fmt.Sprintf("%f", r.RampUpTime),
			"BUS_FACTOR" : fmt.Sprintf("%f", r.BusFactor),
			"CORRECTNESS" : fmt.Sprintf("%f", r.Correctness),
			"LICENSE" : fmt.Sprintf("%f", r.LicenseCompatibility),
			"DEPENDENCIES" : fmt.Sprintf("%f", r.Dependency),
			"PULL_REQ_LOC" : fmt.Sprintf("%f", r.LocPRCR),
		},
	}

	if _, err := o.Update(ctx, objectAttrsToUpdate); err != nil {
		return fmt.Errorf("ObjectHandle(%q).Update: %v", object, err)
	}

	return nil
}

File: src/gcp_calc/module_rate.py
import os
import re
from google.cloud import storage

os.environ["GCLOUD_PROJECT"] = "ece461-module-registry"
bucket_name = "tmr-bucket"

def rating_init():
    # Connects to storage client
    storage_client = storage.Client()

    blobs = storage_client.list_blobs(bucket_name)

    for blob in blobs:

        metageneration_match_precondition = blob.metageneration

        metadata = blob.metadata

        metadata['NET_SCORE'] = 0.0
        metadata['RESPONSIVE_MAINTAINER'] = 0.0
        metadata['RAMP_UP'] = 0.0
        metadata['BUS_FACTOR'] = 0.0
        metadata['CORRECTNESS'] = 0.0
        metadata['LICENSE'] = 0.0
        metadata['DEPENDENCIES'] = 0.0
        metadata['PULL_REQ_LOC'] = 0.0

        blob.metadata = metadata
        blob.patch(if_metageneration_match=metageneration_match_precondition)

rating_init()